k8s/refs/broker_getter_func.go

```diff
@@ -0,0 +1,29 @@
+package refs
+
+import (
+	"github.com/deis/steward-framework/k8s/data"
+	"github.com/deis/steward-framework/k8s/restutil"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/rest"
+)
+
+// BrokerGetterFunc is the function that attempts to retrieve a broker at the given object ref
+type BrokerGetterFunc func(api.ObjectReference) (*data.Broker, error)
+
+// NewK8sBrokerGetterFunc returns a BrokerGetterFunc  backed by a real kubernetes client
+func NewK8sBrokerGetterFunc(restIface rest.Interface) BrokerGetterFunc {
+	return func(ref api.ObjectReference) (*data.Broker, error) {
+		ret := new(data.Broker)
+		url := restutil.AbsPath(
+			restutil.APIVersionBase,
+			restutil.APIVersion,
+			false,
+			ref.Namespace,
+			ref.Name,
+		)
```

Comment:
This looks off... I think the pluralized resource type needs to appear in the path. Also, I think that in the call to `restutil.AbsPath(...)`, you are passing `ref.Name` to the argument which is expected (per the function signature) to be the aforementioned `pluralName`.

I believe the following change would correct both these problems:

```
url := append(
  restutil.AbsPath(
    restutil.APIVersionBase,
    restutil.APIVersion,
    false,
    ref.Namespace,
    data.BrokerKindPlural,
  ),
  ref.Name,
)
```

Similar issue exists with the other types.

Alternatively, `restutil.AbsPath(...)` could be refactored. I leave it up to you what you think is the best way to solve for this.